### PR TITLE
Badges

### PR DESCRIPTION
**Pull request by SonyaPodliesna**
Create Badges page. Add modals for Dashboard.

### CODE REVIEW

File: src/app/badges/badges.component.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  templateUrl: './badges.component.html',
+  styleUrls: ['./badges.component.scss']
+})
+export class BadgesComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
```
Comment (tgogina) [inline src/app/badges/badges.component.ts]: Please remove empty constructor and ngOnInit

File: src/app/badges/components/full-list-achievements/full-list-achievements.component.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    { pic: './assets/achievement-icon.png', name: 'Exoft Corporate Power', xp: 20, dateEvent: new Date(2021, 2, 31, 12, 0, 0) },
+    { pic: './assets/achievement-icon.png', name: 'Exoft Corporate Power', xp: 20, dateEvent: new Date(2021, 2, 31, 12, 0, 0) }
+  ];
+
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
```
Comment (tgogina) [inline src/app/badges/components/full-list-achievements/full-list-achievements.component.ts]: Please remove empty constructor and ngOnInit

File: src/app/badges/components/total-achievements/total-achievements.component.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    { pic: './assets/achievement-icon.png', name: 'Exoft Turbo Power', xp: 15 },
+    { pic: './assets/achievement-icon.png', name: 'Exoft Turbo Power', xp: 15 },
+    { pic: './assets/achievement-icon.png', name: 'Exoft Skylark Power', xp: 20 },
+    { pic: './assets/achievement-icon.png', name: 'Exoft Corpo Power', xp: 20 }
+  ];
+
+  totalXP = 0;
+
+  constructor() { }
+
```
Comment (tgogina) [inline src/app/badges/components/total-achievements/total-achievements.component.ts]: Please remove empty constructor

File: src/app/badges/components/user/user.component.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  styleUrls: ['./user.component.scss']
+})
+export class UserComponent implements OnInit {
+  user = {userPic: '../assets/user-pic.jpg', userFirstName: 'Sonya', userLastName: 'Podliesna'};
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
```
Comment (tgogina) [inline src/app/badges/components/user/user.component.ts]: empty constructor and ngOnInit :)
please double check all files and remove the unused ones

File: src/app/dashboard/components/personal-achievements/personal-achievements.component.ts [hunk 5]
```diff
@@ -1,15 +1,26 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
```
Comment (tgogina) [inline src/app/dashboard/components/personal-achievements/personal-achievements.component.ts]: you do not implement OnInit here, so pleasse remove it from import

File: src/app/shared/components/thank-you/thank-you.component.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import { MatDialog } from '@angular/material/dialog';
+import { OtherUserProfileComponent } from '../../modals/other-user-profile/other-user-profile.component';
+
+@Component({
+  selector: 'app-thank-you',
+  templateUrl: './thank-you.component.html',
+  styleUrls: ['./thank-you.component.scss']
+})
+export class ThankYouComponent implements OnInit {
+  public user = { userPic: './assets/user-pic.jpg', firstName: 'Taras', lastName: 'Sava', userXP: 15 };
```
Comment (tgogina) [inline src/app/shared/components/thank-you/thank-you.component.ts]: `public` is default access modifier. you did not provide it in all other places, so i believe you can remove it here too :)

File: src/app/shared/material/material.module.ts [hunk 7]
```diff
@@ -16,6 +16,9 @@ import { MatIconModule } from '@angular/material/icon';
 import { MatTooltipModule } from '@angular/material/tooltip';
 import { MatCardModule } from '@angular/material/card';
 import { MatDividerModule } from '@angular/material/divider';
+import { MatDialogModule } from '@angular/material/dialog';
+import { MatSelectModule } from '@angular/material/select';
+import { MatInputModule } from '@angular/material/input';
 
 
 @NgModule({
```
Comment (tgogina) [inline src/app/shared/material/material.module.ts]: You have different amount of empty lines after imports (in some files - 1 line, on other - 2). Please make sure you have same amount of empty lines after imports in every ts file

File: src/app/badges/badges.component.ts (hunk 1 above)
Comment (SonyaPodliesna) [reply src/app/badges/badges.component.ts]: fixed

File: src/app/badges/components/full-list-achievements/full-list-achievements.component.ts (hunk 2 above)
Comment (SonyaPodliesna) [reply src/app/badges/components/full-list-achievements/full-list-achievements.component.ts]: fixed

File: src/app/badges/components/total-achievements/total-achievements.component.ts (hunk 3 above)
Comment (SonyaPodliesna) [reply src/app/badges/components/total-achievements/total-achievements.component.ts]: fixed

File: src/app/badges/components/user/user.component.ts (hunk 4 above)
Comment (SonyaPodliesna) [reply src/app/badges/components/user/user.component.ts]: fixed

File: src/app/dashboard/components/personal-achievements/personal-achievements.component.ts (hunk 5 above)
Comment (SonyaPodliesna) [reply src/app/dashboard/components/personal-achievements/personal-achievements.component.ts]: fixed

File: src/app/shared/components/thank-you/thank-you.component.ts (hunk 6 above)
Comment (SonyaPodliesna) [reply src/app/shared/components/thank-you/thank-you.component.ts]: fixed

File: src/app/shared/material/material.module.ts (hunk 7 above)
Comment (SonyaPodliesna) [reply src/app/shared/material/material.module.ts]: fixed

**Review by tgogina — APPROVED** [review]

**Pull request merged**